IDEOorg/steps-bot
# #164700316 As a bot, I should be able to send text messages with any credentials via Twilio

### What does this PR do?
Enable bot to use orgs credentials

### Description of tasks to be completed
- Remove botkit implementation for receiving sms
- Create custom implementation for receiving sms
- Enable bot to fetch org credentials and use it for sending sms
- Add node clusters to enhance bot performance

### How should this be manually tested?
* Clone this repo
* Checkout to a branch `ft-use-orgs-twilio-sms-credentials-164700316`.

* Run `npm start` and `npm run start:live:dev`
* Setup Twilio webhook.
* Log in as a coach and create a client if not already present.
* Add a task to the client's work plan.
* Visit [twilio-gui](http://twilio-gui.herokuapp.com) and get a phone number.
* login to `twilio-gui`
* Using the direct message feature from the coach app, send a message to the client
* Notice that the client did not receive the message.

**Test For When the credential is provided**
* Update your local org account with the available Twilio credentials.
* Using the direct message feature from the coach app, send a message to the client
* Notice that the client receives the message.

**Test For Development Usage**
* Add `NODE_ENV=development` to the local `.env` file.
* Using the direct message feature from the coach app, send a message to the client
* Notice that the client receives the message.

### What are the relevant pivotal tracker stories?
[#164700316](https://www.pivotaltracker.com/story/show/164700316)

### Any Background Context
"Right now, IDEO.org is paying for every text sent to/from Roo. Our plan is to open the app up to the public, and that means a lot more texts...and a lot more money. But we don't have that money!

in the public version, we are asking admins to create their own Twilio account and to enter their Twilio credentials in their org settings (saved to the Org table in the DB).

The bot should use the saved credentials for any text message sent to/from a client of an organization. Here is an example.

> Tracy is a client of ABC Financial Services in Texas. When Roo sends Tracy a text, it uses the Twilio credentials for ABC Financial Services, and they receive their own bill at the end of the month.

> Joseph is a client of XYZ Financial Services in California. When Roo sends Joseph a text, it uses the Twilio credentials for XYZ Financial Services, and they will receive their own bill at the end of the month.
" - @matthewepler 

### Screenshots?
N/A


### Questions
N/A
